api/_modules/user/user.service.ts

```diff
@@ -0,0 +1,79 @@
+import jwt from 'jsonwebtoken'
+
+import { config } from '../../_utils/config'
+import { logger } from '../../_utils/logger'
+import { mailer } from '../../_utils/mailer'
+import { UserRepository } from './user.repository'
+
+export class UserService {
+  private readonly userRepository: UserRepository
+
+  constructor() {
+    this.userRepository = new UserRepository()
+  }
+
+  async registerOrSignInUser(email: string) {
+    logger.debug('register or sign in user', {
+      service: 'user.service.registerOrSignInUser',
+      meta: { email }
+    })
+
+    let user: any = await this.userRepository.findByEmail(email)
+
+    if (!user) {
+      logger.debug('didnt find any user in database', {
+        service: 'user.service.registerOrSignInUser'
+      })
+
+      user = await this.userRepository.createUser(email)
+    }
+
+    logger.debug('found user in database', {
+      service: 'user.service.registerOrSignInUser',
+      meta: { user }
+    })
+
+    await this.sendConfirmAccountEmail(user)
+
+    return user
+  }
+
+  async sendConfirmAccountEmail(user: any) {
```

Comment:
Need to create a user type

```suggestion
  async sendConfirmAccountEmail(user: UserType) {
```